spark/v3.3/spark/src/test/java/org/apache/iceberg/TestTableSerialization.java

```diff
@@ -33,11 +33,25 @@
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
 
+@RunWith(Parameterized.class)
 public class TestTableSerialization {
 
+  public TestTableSerialization(String isObjectStoreEnabled) {
+    this.isObjectStoreEnabled = isObjectStoreEnabled;
+  }
+
+  @Parameterized.Parameters(name = "isObjectStoreEnabled = {0}")
+  public static Object[] parameters() {
+    return new Object[] {"true", "false"};
```

Comment:
Do we need to parameterize? I suppose it's fine because there's very few tests in the class, and they're all pretty lightweight so the runtime of the tests won't increase much. I was thinking we can just add a separate test where object store is set to true

Reply:
> Do we need to parameterize ?

just wanted to have the same coverage for ObjectStoreLocation provider like DefaultLocationProvider and avoid duplication, hence made it parametrized, to test serialization of all the table type's transactionTable / metadataTable / normal table. 

Please let me know you thoughts, happy to make the changes accordingly.

Reply:
Side note : presently it's just testing serialization with kryo serializer, imho we should have tests for java serializer as well, happy to add these as well, if we all agree.